File: DoItTest/ClientApp/src/domain/tests/items/textFieldItem.ts
import { mapTextFildAnswerOption, TextFildAnswerOption } from "./answerOption/textFildAnswerOption";
import { TestItem } from "./testItem";
import { TestItemType } from "./testItemType";

export class TextFieldItem extends TestItem {
    constructor(
        public id: string,
        public testId: string,
        public type: TestItemType,
        public question: string,
        public imageBase64: string | null,
        public answerOption: TextFildAnswerOption
    ) {
        super(id, testId, type, question, imageBase64)
    }
}

export function mapToTextFieldItem(value: any): TextFieldItem {
    const answerOption = mapTextFildAnswerOption(value.answerOption)

    return new TextFieldItem(value.id, value.testId, value.type, value.question, value.imageBase64, answerOption)
}

File: DoItTest.Domain/Tests/TestItems/TextFieldItem.cs
using DoItTest.Domain.Tests.TestItems.AnswerOptions;

namespace DoItTest.Domain.Tests.TestItems
{
    public class TextFieldItem : TestItem
    {
        public TextFildAnswerOption AnswerOption { get; }

        public TextFieldItem(Guid id, Guid testId, TestItemType type,
            String question, String? imageBase64, TextFildAnswerOption answerOption) 
            : base(id, testId, type, question, imageBase64)
        {
            AnswerOption = answerOption;
        }
    }
}
